File: Lab_1/src/backend/Triangle.php
<?php

require_once "Shape.php";

class Triangle extends Shape
{
  private float $x;
  private float $y;
  private bool $low_quadrant = FALSE;

  public function __construct(int $quadrant, float $x, float $y){
    parent::__construct($quadrant);
    $this->x = $x;
    $this->y = $y;
  }  

  public function isInShape(float $x, float $y): bool
  {
    if ($quadrant == 3 || $quadrant == 4) {
      $low_quadrant = TRUE;
    } 
    if ($low_quadrant){
      return $this->isInQuadrant($x, $y) && ($y >= ( -2 * $x - $this->x));
    } else {
      return $this->isInQuadrant($x, $y) && ($y <= ( -2 * $x + $this->x));
    }
  }

}

File: Lab_2/src/main/java/Shapes/Triangle.java
package Shapes;
public class Triangle extends AbstractShape{

  private double x;
  private double y;
  private boolean low_quadrant = false;

  public Triangle( Integer quadrant, double x, double y ) 
  {
    super(quadrant);
    this.x = x;
    this.y = y;
    if ( this.quadrant == 3 || this.quadrant == 4) { this.low_quadrant = true; } 
  }

  @Override
  public boolean isInShape(double x, double y)
  {
    if ( this.low_quadrant ){
      return this.isInQuadrant(x, y) && (y >= ( ( -this.y / this.x ) * x + this.y));
    } else {
      return this.isInQuadrant(x, y) && (y <= ( ( -this.y / this.x ) * x + this.y));
    }
  }
}